### PR TITLE
Feature/new rabbit bind params

### PR DESCRIPTION
**Pull request by ianrocha**
# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #572 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@ianrocha thanks for this PR, there are conflicts that must be resolved.

**Comment by yaron2** [comment]
Make sure you started on a clean branch and that you fetched the latest from master.

**Comment by yaron2** [comment]
There's Go linter issues: https://github.com/dapr/components-contrib/pull/573/checks?check_run_id=1653080393

**Comment by ianrocha** [comment]
Hi @yaron2, sorry for the fails, but can you help my with this one?

https://github.com/dapr/components-contrib/pull/573/checks?check_run_id=1653544208

Do you have any clue about what a could do to solve this?

**Comment by yaron2** [comment]
> Hi @yaron2, sorry for the fails, but can you help my with this one?
> 
> https://github.com/dapr/components-contrib/pull/573/checks?check_run_id=1653544208
> 
> Do you have any clue about what a could do to solve this?

Yeah it's just a format issue. For utils/metadata.go, make sure you run go fmt -s utils/metadata.go

**Comment by ianrocha** [comment]
@yaron2 thanks! it worked great 👍

**Comment by yaron2** [comment]
Just to make it clear, this PR adds dead lettering to both the binding AND pubsub?

**Comment by ianrocha** [comment]
No, the dead lettering arguments are only for the binding. 
The change I made for the pubsub is that now it will receive the parameter exchangeKind, before it could only create an exchange of kind "fanout".

**Comment by yaron2** [comment]
> No, the dead lettering arguments are only for the binding.
> The change I made for the pubsub is that now it will receive the parameter exchangeKind, before it could only create an exchange of kind "fanout".

Thanks for clarifying!

Did you perhaps get to test the RabbitMQ pub/sub and bindings after this change? to make sure there are no regressions.

**Comment by ianrocha** [comment]
It was working good before I merged with master, now I need to test again xD... I think I'll be able to do this on the weekend.

**Comment by yaron2** [comment]
hey @ianrocha, please resolve the conflict and we can go ahead and merge this. did you get to test by any chance?

**Review by yaron2 — APPROVED** [review]
LGTM

File: pubsub/rabbitmq/metadata.go [hunk 1]
```diff
@@ -40,6 +41,12 @@ func createMetadata(pubSubMetadata pubsub.Metadata) (*metadata, error) {
 		return &result, fmt.Errorf("%s missing RabbitMQ consumerID", errorMessagePrefix)
 	}
 
+	if val, found := pubSubMetadata.Properties[metadataExchangeKind]; found && val != "" {
+		result.exchangeKind = val
+	} else {
+		return &result, fmt.Errorf("%s missing RabbitMQ exchangeKind", errorMessagePrefix)
```
Comment (yaron2) [inline pubsub/rabbitmq/metadata.go:47]: Please set the exchangeKind here to "fanout" by default

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
@ianrocha I resolved the conflict, left one comment.

**Comment by yaron2** [comment]
ping @ianrocha

**Comment by yaron2** [comment]
@ianrocha pinging again.

**Comment by yaron2** [comment]
Closing as author is non-responsive

**Pull request closed without merging**